rust/operator-binary/src/crd/utils.rs

```diff
@@ -0,0 +1,104 @@
+use std::{borrow::Cow, collections::HashMap, num::TryFromIntError};
+
+use futures::future::try_join_all;
+use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_operator::{
+    commons::listener::Listener, k8s_openapi::api::core::v1::Pod,
+    kube::runtime::reflector::ObjectRef, utils::cluster_info::KubernetesClusterInfo,
+};
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("unable to get {listener} (for {pod})"))]
+    GetPodListener {
+        source: stackable_operator::client::Error,
+        listener: ObjectRef<Listener>,
+        pod: ObjectRef<Pod>,
+    },
+
+    #[snafu(display("{listener} (for {pod}) has no address"))]
+    PodListenerHasNoAddress {
+        listener: ObjectRef<Listener>,
+        pod: ObjectRef<Pod>,
+    },
+
+    #[snafu(display("port {port} ({port_name:?}) is out of bounds, must be within {range:?}", range = 0..=u16::MAX))]
```

Comment:
```suggestion
    #[snafu(display("port {port} ({port_name:?}) is out of bounds, must be within {range:?}", range = 0..=u16::MAX))]
```
This works for ports < 1024?